app/src/main/java/ru/suslovalex/androidacademyapp/worker/MoviesWorker.kt

```diff
@@ -0,0 +1,41 @@
+package ru.suslovalex.androidacademyapp.worker
+
+import android.content.Context
+import android.util.Log
+import androidx.core.content.ContentProviderCompat
+import androidx.work.*
+import org.koin.core.component.KoinApiExtension
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+import ru.suslovalex.androidacademyapp.data.Movie
+import ru.suslovalex.androidacademyapp.repository.MoviesRepository
+
+
+@KoinApiExtension
+class MoviesWorker(
+    context: Context,
+    workerParams: WorkerParameters,
+) : CoroutineWorker(context, workerParams), KoinComponent {
+
+    private val moviesRepository: MoviesRepository by inject()
+
+    override suspend fun doWork(): Result {
+        return try {
+            val movies: List<Movie> = moviesRepository.loadUpcomingMovies()
+            deleteData()
```

Comment:
@AleksandrAlex Why do we clean database? We just only need to add new data from network. Remove this code from here.